tst/unit/test_metadata.cpp

```diff
@@ -30,14 +30,56 @@ TEST_CASE("Built-in flags are registered", "[Metadata]") {
 TEST_CASE("A Metadata flag is allocated", "[Metadata]") {
     GIVEN("A User Flag") {
         auto const f = Metadata::AllocateNewFlag("TestFlag");
-        // Note: `parthenon::internal` is subject to change, and so this test may rightfully break
-        // later - this test needn't be maintained if so.
+        // Note: `parthenon::internal` is subject to change, and so this test may
+        // rightfully break later - this test needn't be maintained if so.
         //
-        // Checks that the first allocated flag is equal to `Max` - the final built-in flag + 1.
-        REQUIRE(f.InternalFlagValue() == static_cast<int>(parthenon::internal::MetadataInternal::Max));
+        // Checks that the first allocated flag is equal to `Max` - the final built-in
+        // flag + 1.
+        REQUIRE(
+            f.InternalFlagValue() ==
+                static_cast<int>(parthenon::internal::MetadataInternal::Max));
         REQUIRE("TestFlag" == f.Name());
 
         // It should throw an error if you try to allocate a new flag with the same name.
         REQUIRE_THROWS_AS(Metadata::AllocateNewFlag("TestFlag"), std::runtime_error);
     }
-}
\ No newline at end of file
+}
+
+TEST_CASE("A Metadata struct is created", "[Metadata]") {
+    GIVEN("A default Metadata struct") {
+        Metadata m;
+
+#define PARTHENON_INTERNAL_FOR_FLAG(name) REQUIRE(!m.IsSet(Metadata::name));
+    PARTHENON_INTERNAL_FOREACH_BUILTIN_FLAG
+#undef PARTHENON_INTERNAL_FOR_FLAG
+    }
+
+    GIVEN("Setting an arbitrary Metadata flag only sets that flag") {
+        Metadata m;
+
+        m.Set(Metadata::FillGhost);
+
+
+#define PARTHENON_INTERNAL_FOR_FLAG(name) \
+    if (Metadata::name != Metadata::FillGhost) REQUIRE(!m.IsSet(Metadata::name));
+    PARTHENON_INTERNAL_FOREACH_BUILTIN_FLAG
+#undef PARTHENON_INTERNAL_FOR_FLAG
+
+        REQUIRE(m.IsSet(Metadata::FillGhost));
+    }
+
+    GIVEN("Two Equivalent Metadata Structs Are Compared") {
+        Metadata a({Metadata::Cell}), b({Metadata::Face});
+
+        b.Unset(Metadata::Face);
+        b.Set(Metadata::Cell);
+
+        REQUIRE(a == b);
+    }
+
+    GIVEN("Two Different Metadata Structs Are Compared") {
```

Comment:
Maybe compare two Metadata with different numbers of flags and/or multiple flags